Fixed lora trainer bug

austin2035 opened the pull request:
fixed lora trainer bug


```bash
base) root@ubuntu22:~/ACE-Step# sh train.sh 
2025-05-15 22:59:11.221 | INFO     | acestep.pipeline_ace_step:load_checkpoint:153 - Load models from: /root/models/
Traceback (most recent call last):
  File "/root/ACE-Step/trainer.py", line 887, in <module>
    main(args)
  File "/root/ACE-Step/trainer.py", line 819, in main
    model = Pipeline(
            ^^^^^^^^^
  File "/root/ACE-Step/trainer.py", line 72, in __init__
    assert lora_config_path is not None, "Please provide a LoRA config path"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Please provide a LoRA config path
(base) root@ubuntu22:~/ACE-Step# vim trainer.py 
(base) root@ubuntu22:~/ACE-Step# sh train.sh 
2025-05-15 23:03:24.187 | INFO     | acestep.pipeline_ace_step:load_checkpoint:153 - Load models from: /root/models/
Traceback (most recent call last):
  File "/root/ACE-Step/trainer.py", line 888, in <module>
    main(args)
  File "/root/ACE-Step/trainer.py", line 819, in main
    model = Pipeline(
            ^^^^^^^^^
  File "/root/ACE-Step/trainer.py", line 79, in __init__
    lora_config = json.load(f)
                  ^^^^
UnboundLocalError: cannot access local variable 'json' where it is not associated with a value
```

Although my command argument provided lora_config_path, the Pipeline instance in the main function did not use it, resulting in an error. I added relevant code to facilitate passing the lora_config_path parameter.

ChuxiJ commented:
JSON has been imported, so it's unnecessary to import it again.
Please check your code again.
The error is that the Lora config path is not input.

austin2035 commented:
> JSON has been imported, so it's unnecessary to import it again. Please check your code again. The error is that the Lora config path is not input.

1. Please check the logs, and after I added the import code, it worked fine. I used the code from this repository entirely.
2. Although my command argument provided `lora_config_path`, the Pipeline instance in the main function did not use it, resulting in an error. I added relevant code to facilitate passing the `lora_config_path` parameter.

woct0rdho commented:
I think it's better to remove the imports here:
https://github.com/ace-step/ACE-Step/blob/6c14becf980793b5cf9e355da5b767132d3df34a/trainer.py#L102-L103

If we import json in the function `__init__`, then Python will think it's a local variable in the function (rather than the global variable), and complain that it's used before assigned.
